scrapper.py

```diff
@@ -21,43 +32,189 @@ class IncorrectNumberOfArticlesError(Exception):
+
+        if text_date:
+            date_re = re.search(r'\d{2}\.\d{2}\.\d{4}', text_date.group(0))
+
+            self.article.date = datetime.datetime.strptime(date_re.group(0), '%d.%m.%Y')
+
+
 class Crawler:
     """
     Crawler implementation
     """
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self._seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
+        self._article_count = 0
 
     def _extract_url(self, article_bs):
-        pass
+        """
+        Finds urls from the given article_bs
+        """
+        table_rows = article_bs.find_all('tr', {"class": "unnrow"})
+
+        for table_row in table_rows:
+            if self._article_count + 1 > self.max_articles:
+                break
+
+            links = table_row.find_all('a')
+
+            for link in links:
+                if self._article_count + 1 > self.max_articles:
+                    break
+
+                match = re.match(r'^\?anum', link['href'])
+
+                if not match:
+                    continue
+
+                self.urls.append(''.join(["http://www.vestnik.unn.ru/ru/nomera", link['href']]))
+                self._article_count += 1
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for seed_url in self._seed_urls:
+            if self._article_count + 1 > self.max_articles:
+                break
+
+            response = requests.get(seed_url)
+            article_bs = BeautifulSoup(response.text, features="html.parser")
+            self._extract_url(article_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self._seed_urls
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    if os.path.isdir(base_path):
+        file_names = os.listdir(base_path)
+
+        for file_name in file_names:
+            os.remove(os.path.join(base_path, file_name))
+    else:
+        os.makedirs(base_path)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r') as crawler_file:
+        crawler_config = json.load(crawler_file)
+
+        max_articles = crawler_config['total_articles_to_find_and_parse']
+
+        if not isinstance(max_articles, int):
+            raise IncorrectNumberOfArticlesError
+
+        if max_articles <= 0:
+            raise IncorrectNumberOfArticlesError
+
+        if max_articles > 100:
+            raise NumberOfArticlesOutOfRangeError
+
+        seed_urls = crawler_config['seed_urls']
+
+        if not isinstance(seed_urls, list) or not seed_urls:
+            raise IncorrectURLError
+
+        for seed_url in seed_urls:
+            match = re.match(r'(^http://|^https://)', seed_url)
```

Comment:
sould you accept such urls as seed ones? https://github.com/fipl-hse/2021-2-level-ctlr/pull/46/files

should you?